Validator/ValidatorWrapper.php

```diff
@@ -0,0 +1,63 @@
+<?php
+
+/*
+ * This file is part of the Sonata Project package.
+ *
+ * (c) Thomas Rabaix <thomas.rabaix@sonata-project.org>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sonata\AdminBundle\Validator;
+
+use Symfony\Component\Validator\Validator\ValidatorInterface;
+use Symfony\Component\Validator\ValidatorInterface as LegacyValidatorInterface;
+
+class ValidatorWrapper implements LegacyValidatorInterface
+{
+    /* @var mixed ValidatorInterface|LegacyValidatorInterface $validator */
+    private $validator;
+
+    /**
+     * ValidatorWrapper constructor.
+     * @param mixed ValidatorInterface|LegacyValidatorInterface  $validator
+     */
+    public function __construct($validator)
+    {
+        if (!$validator instanceof ValidatorInterface && !$validator instanceof LegacyValidatorInterface) {
+            throw new \InvalidArgumentException('Argument 1 must be an instance of Symfony\Component\Validator\Validator\ValidatorInterface or Symfony\Component\Validator\ValidatorInterface');
+        }
+
+        $this->validator = $validator;
+    }
+
+    public function validate($value, $groups = null, $traverse = false, $deep = false)
+    {
+        $this->validator->validate($value, $groups, $traverse, $deep);
+    }
+
+    public function validateProperty($containingValue, $property, $groups = null)
+    {
+        $this->validator->validateProperty($containingValue, $property, $groups);
+    }
+
+    public function validatePropertyValue($containingValue, $property, $value, $groups = null)
+    {
+        $this->validator->validatePropertyValue($containingValue, $property, $value);
+    }
+
+    public function validateValue($value, $constraints, $groups = null)
+    {
+        return $this->validator->validateValue($value, $constraints, $groups);
+    }
+
+    public function getMetadataFactory()
+    {
+        if (method_exists($this->validator, 'getMetadataFor')) {
+            return $this->validator->getMetadataFor($this->getClass());
+        }
+
+        return $this->validator->getMetadataFactory()->getMetadataFor($this->getClass());
+    }
+}
```

Comment:
Please configure your editor to add a newline at end of file, it will make this pictogram disappear